### PR TITLE
Add support for flashing bootloader from ROMFS via mavlink command

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by davidbuzz** [comment]
quick review here looks good, I didn't test tho.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -2456,6 +2456,20 @@ void GCS_MAVLINK::send_simstate() const
 #endif
 }
 
+MAV_RESULT GCS_MAVLINK::handle_command_flash_bootloader(const mavlink_command_long_t &packet)
+{
+    if (int(packet.param5) != 290876) {
```
Comment (khancyr) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: uint32_t ?

Comment (tridge) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: fixed

**Comment by peterbarker** [comment]
On Wed, 27 Jun 2018, Andrew Tridgell wrote:

> In libraries/GCS_MAVLink/GCS_Common.cpp:
> 
> > @@ -2456,6 +2456,20 @@ void GCS_MAVLINK::send_simstate() const
>  #endif
>  }
> 
> +MAV_RESULT GCS_MAVLINK::handle_command_flash_bootloader(const mavlink_comma
> nd_long_t &packet)
> +{
> +    if (int(packet.param5) != 290876) {
> 
> fixed

Now we have two magic values that work :-)

**Pull request merged**